lib/senders/sms.js

```diff
@@ -84,17 +84,28 @@ module.exports = function (log, translator, templates, smsConfig) {
     return P.promisify(object[methodName], { context: object })
   }
 
-  function getMessage (templateName, acceptLanguage) {
+  function getMessage (templateName, acceptLanguage, signinCode) {
     var template = templates['sms.' + templateName]
 
     if (! template) {
       log.error({ op: 'sms.getMessage.error', templateName: templateName })
       throw error.invalidMessageId()
     }
 
+    var link
```

Comment:
The route handler feels like the wrong layer of abstraction to create the link at to me. `lib/senders/sms` already does stuff with `config.sms` and is responsible for everything to do with the message itself.

I get your point about localizing cause and effect, but we already have logic split between these two modules and this seems the more appropriate of the two.

If it's something you feel strongly about though, I'm happy to do it how you want.